tools/astarte_e2e/lib/astarte_e2e/config.ex

```diff
@@ -175,6 +220,29 @@ defmodule AstarteE2E.Config do
     ]
   end
 
+  def service_notifier_config do
+    with {:ok, mail_adapter} when not is_nil(mail_adapter) <- mail_service(),
+         {:ok, base_uri} when not is_nil(base_uri) <- mail_api_base_uri(),
+         {:ok, domain} when not is_nil(domain) <- mail_domain(),
+         {:ok, api_key} when not is_nil(api_key) <- mail_api_key(),
+         {:ok, from} when not is_nil(from) <- mailer_from_address(),
+         {:ok, to} when not is_nil(to) <- mailer_to_address() do
+      %{
+        chained_adapter: mail_adapter,
+        api_key: api_key,
+        domain: domain,
+        base_uri: base_uri
+      }
+    else
+      _ ->
+        Logger.warn("Incomplete mail configuration. The Local Adapter will be used.",
+          tag: "local_adapter_fallback"
+        )
```

Comment:
This makes sense, but I would not print it if `mail_service` is `nil` (since otherwise E2E without a configured mailer becomes particularly spammy). It's ok to print this with incomplete configuration though.